joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/JavaSrc2Cpg.scala

```diff
@@ -29,25 +33,66 @@ class JavaSrc2Cpg extends X2CpgFrontend[Config] {
   def createCpg(config: Config): Try[Cpg] = {
     withNewEmptyCpg(config.outputPath, config: Config) { (cpg, config) =>
       new MetaDataPass(cpg, language, config.inputPath).createAndApply()
-      val sourcesInfo = getSourcesFromDir(config)
+      val dependencies        = getDependencyList(config)
+      val hasLombokDependency = dependencies.exists(_.contains("lombok"))
+      val sourcesInfo         = getSourcesFromDir(config, hasLombokDependency)
       if (sourcesInfo.sourceFiles.isEmpty) {
         logger.error(s"no source files found at path ${config.inputPath}")
       } else {
         logger.info(s"found ${sourcesInfo.sourceFiles.size} source files")
       }
 
-      val astCreator = new AstCreationPass(sourcesInfo.typeSolverSourceDir, sourcesInfo.sourceFiles, config, cpg)
+      val astCreator =
+        new AstCreationPass(sourcesInfo, config, cpg, dependencies)
       astCreator.createAndApply()
       new ConfigFileCreationPass(config.inputPath, cpg).createAndApply()
       new TypeNodePass(astCreator.global.usedTypes.keys().asScala.toList, cpg)
         .createAndApply()
     }
   }
 
+  private def getDependencyList(config: Config): Seq[String] = {
+    val codeDir = config.inputPath
+    if (config.fetchDependencies) {
+      DependencyResolver.getDependencies(Paths.get(codeDir)) match {
+        case Some(deps) => deps.toSeq
+        case None =>
+          logger.warn(s"Could not fetch dependencies for project at path $codeDir")
+          Seq()
+      }
+    } else {
+      logger.info("dependency resolving disabled")
+      Seq()
+    }
+  }
+
+  private def getDelombokMode(config: Config): DelombokMode = {
+    config.delombokMode.map(_.toLowerCase) match {
+      case None                 => DelombokMode.Default
+      case Some("no-delombok")  => DelombokMode.NoDelombok
+      case Some("default")      => DelombokMode.Default
+      case Some("types-only")   => DelombokMode.TypesOnly
+      case Some("run-delombok") => DelombokMode.RunDelombok
+      case Some(value) =>
+        logger.warn(s"Found unrecognised delombok mode `$value`. Using default instead.")
+        DelombokMode.Default
+    }
+  }
+
   /** JavaParser requires that the input path is a directory and not a single source file. This is inconvenient for
     * small-scale testing, so if a single source file is created, copy it to a temp directory.
     */
-  private def getSourcesFromDir(config: Config): SourceDirectoryInfo = {
+  private def getSourcesFromDir(config: Config, hasLombokDependency: Boolean): SourceDirectoryInfo = {
+    val delombokMode = getDelombokMode(config)
+    if (hasLombokDependency) {
+      logger.info(s"Analysing delomboked code as lombok dependency was found.")
```

Comment:
probably should move that into the match below, otherwise it might seem we're ignoring the command-line choice

Reply:
Good catch!